logstash-plugins/logstash-input-java_input_example
# Avoid Logstash's classes in plugin's uber jar due to shading

## Release notes

[rn:skip]

## What does this PR do?


Fixes the plugin to avoid to pack also the classes of Logstash and log4j.
The example plugin originally used to create a shadow jar with all the dependencies merged into. The dependencies that ends into are also the Logstash classes and the log4j one, which are already provided by the Logstash project itself. This could lead to classpath trashing.
This PR avoid to use shadow the dependencies and leverage the fact that the gem could pack also the jars. 
Furthermore adds a MANIFEST, in the plugin's main jar, to mark it as entry point to be recognized by Logstash. This PR to properly works needs a Logstash version with the https://github.com/elastic/logstash/pull/13979.

## Why is it important/What is the impact to the user?



## Checklist




## Author's Checklist


- [x] test locally installing the gem file

## How to test this PR locally
- after populate the `LOGSTASH_CORE_PATH` property in `./gradle.properties`
- verify the tests with `./gradlew test` and build the gem file `./gradlew gem`
- install the gem into a Logstash with:
```sh
bin/logstash-plugin install --local /path/to/gem/logstash-input-java_input_example-1.0.2.gem
```
- run Logstash pipeline that uses it:
```
input {	
  java_input_example {
    count => 2
  }
}

output {
  stdout {
    codec => rubydebug
  }
}
```

## Related issues


- Related https://github.com/elastic/logstash/pull/13979

## Use cases



## Screenshots



## Logs
